Player.java

```diff
@@ -17,8 +17,10 @@ public Player( String name, int dmg, int hp, double as, int xppl, int dmgipl, in
     }
 
     public void attack(Unit enemy) {
-        super.attack(enemy);
-        this.xp = this.xp+super.getDMG();
+        if (enemy.isAlive() && this.isAlive()) {
+            enemy.setHP(enemy.getHp()-this.getDMG());
+        }
+        this.xp = this.xp+this.getDMG();
```

Comment:
Annyi van, hogy ha 5 elete van, en 10-et tudnek sebezni, akkor is csak 5 sebzest viszek be.